runtime/src/mm/cpp/Memory.cpp

```diff
@@ -75,3 +93,57 @@ extern "C" RUNTIME_NOTHROW void ClearTLS(MemoryState* memory) {
 extern "C" RUNTIME_NOTHROW ObjHeader** LookupTLS(void** key, int index) {
     return mm::ThreadRegistry::Instance().CurrentThreadData()->tls().Lookup(key, index);
 }
+
+extern "C" RUNTIME_NOTHROW void* CreateStablePointer(ObjHeader* object) {
+    auto* threadData = mm::ThreadRegistry::Instance().CurrentThreadData();
+    return mm::StableRefRegistry::Instance().RegisterStableRef(threadData, object);
+}
+
+extern "C" RUNTIME_NOTHROW void DisposeStablePointer(void* pointer) {
+    auto* threadData = mm::ThreadRegistry::Instance().CurrentThreadData();
+    auto* node = static_cast<mm::StableRefRegistry::Node*>(pointer);
+    mm::StableRefRegistry::Instance().UnregisterStableRef(threadData, node);
+}
+
+extern "C" RUNTIME_NOTHROW OBJ_GETTER(DerefStablePointer, void* pointer) {
+    auto* node = static_cast<mm::StableRefRegistry::Node*>(pointer);
+    ObjHeader* object = **node;
+    RETURN_OBJ(object);
+}
+
+extern "C" RUNTIME_NOTHROW OBJ_GETTER(AdoptStablePointer, void* pointer) {
+    auto* threadData = mm::ThreadRegistry::Instance().CurrentThreadData();
+    auto* node = static_cast<mm::StableRefRegistry::Node*>(pointer);
+    ObjHeader* object = **node;
+    mm::StableRefRegistry::Instance().UnregisterStableRef(threadData, node);
```

Comment:
`object` is probably not a local root here. So it would be more safe to assign it to "return slot" before unregistering the stable ref.

Reply:
Good point, thanks!